rotkehlchen/db/dbhandler.py

```diff
@@ -1344,38 +1342,40 @@ def _get_address_details_if_time(
                 f'Found undecodeable json {result[0][0]} in the DB for {address}.'
                 f'Error: {str(e)}',
             )
-            return None
+            return None, last_update_ts
 
         if not isinstance(json_ret, dict):
             # This can happen if the DB is old and still has only a list of saved tokens
-            return None  # In that case just consider it outdated
+            return None, last_update_ts  # In that case just consider it outdated
 
-        return json_ret
+        return json_ret, last_update_ts
 
-    def get_tokens_for_address_if_time(
+    def get_tokens_for_address(
             self,
             cursor: 'DBCursor',
             address: ChecksumEthAddress,
-            current_time: Timestamp,
-    ) -> Optional[List[EthereumToken]]:
+    ) -> Tuple[Optional[List[EthereumToken]], Optional[Timestamp]]:
         """Gets the detected tokens for the given address if the given current time
         is recent enough.
 
         If not, or if there is no saved entry, return None
         """
-        json_ret = self._get_address_details_if_time(cursor, address, current_time)
+        json_ret, last_update_ts = self._get_address_details(cursor, address)
         if json_ret is None:
-            return None
+            return None, last_update_ts
         tokens_list = json_ret.get('tokens', None)
         if tokens_list is None:
-            return None
+            return None, last_update_ts
 
         if not isinstance(tokens_list, list):
             # This should never happen
             self.msg_aggregator.add_warning(
                 f'Found non-list tokens_list {json_ret} in the DB for {address}.',
             )
-            return None
+            return None, last_update_ts
+
+        if len(tokens_list) == 0:
```

Comment:
This is not what I asked you to do though You just moved the check but same amount of checks will happen each time. https://github.com/rotki/rotki/pull/4536#discussion_r928169962

But I think for now it's fine. I was thinking of making sure to write null in case of empty list, but looking at the table and how it's used I see we have a dictionary entry there that is never null, with multiple keys, so don't wanna touch it now.

Reply:
Yeah, I basically didn't find a way to get rid of this check